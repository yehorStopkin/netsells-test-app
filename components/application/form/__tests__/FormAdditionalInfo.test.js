const { shallowMount } = require('@vue/test-utils');
const FormAdditionalInfo = require('@/components/application/form/FormAdditionalInfo').default;

describe('FormAdditionalInfo', () => {
    getValue = (liveInUk = null, gitProfile = '', aboutYou = '') => {
        return {
            liveInUk,
            gitProfile,
            aboutYou,
        };
    };

    test('component default', () => {
        const wrapper = shallowMount(FormAdditionalInfo, { propsData: { value: getValue(), step: 1 } });
        expect(wrapper.exists()).toBeTruthy();
        expect(wrapper.element).toMatchSnapshot();
    });

    test('component liveInUk', () => {
        const wrapper = shallowMount(FormAdditionalInfo, { propsData: { value: getValue(true), step: 1 } });
        expect(wrapper.exists()).toBeTruthy();
        expect(wrapper.element).toMatchSnapshot();
    });

    test('component gitProfile filled', () => {
        const wrapper = shallowMount(FormAdditionalInfo, { propsData: { value: getValue(false, 'test git profile'), step: 1 } });
        expect(wrapper.exists()).toBeTruthy();
        expect(wrapper.element).toMatchSnapshot();
    });

    test('component aboutYou filled', () => {
        const wrapper = shallowMount(FormAdditionalInfo, { propsData: { value: getValue(false, '', 'test about you'), step: 1 } });
        expect(wrapper.exists()).toBeTruthy();
        expect(wrapper.element).toMatchSnapshot();
    });
});
