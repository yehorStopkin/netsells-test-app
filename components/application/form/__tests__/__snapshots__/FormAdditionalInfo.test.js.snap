// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormAdditionalInfo component aboutYou filled 1`] = `
<div
  class="form-additional-info"
>
  <form-header-stub
    class="form-additional-info__header"
    title="More About You"
  />
   
  <div
    class="form-additional-info__field"
  >
    <div
      class="form-additional-info__field-title required"
    >
      <label
        for="liveInUk"
      >
        Do you live in the UK?
      </label>
    </div>
     
    <el-select
      id="liveInUk"
      popper-class="form-additional-info__field-popper"
    >
      <el-option
        label="Yes"
        value="true"
      />
      <el-option
        label="No"
      />
    </el-select>
  </div>
   
  <div
    class="form-additional-info__field"
  >
    <div
      class="form-additional-info__field-title required"
    >
      <label
        for="gitProfile"
      >
        Your Github Profile
      </label>
    </div>
     
    <div
      id="gitProfile"
      value=""
    />
  </div>
   
  <div
    class="form-additional-info__field"
  >
    <div
      class="form-additional-info__field-title required"
    >
      <label
        for="aboutYou"
      >
        About You
      </label>
    </div>
     
    <div
      id="aboutYou"
      placeholder="Let us know more about you. What are you in to?"
      resize="none"
      rows="9"
      type="textarea"
      value="test about you"
    />
  </div>
   
  <div
    class="form-additional-info__actions"
  >
    <div
      text="NEXT"
      width="115"
    />
  </div>
</div>
`;

exports[`FormAdditionalInfo component default 1`] = `
<div
  class="form-additional-info"
>
  <form-header-stub
    class="form-additional-info__header"
    title="More About You"
  />
   
  <div
    class="form-additional-info__field"
  >
    <div
      class="form-additional-info__field-title required"
    >
      <label
        for="liveInUk"
      >
        Do you live in the UK?
      </label>
    </div>
     
    <el-select
      id="liveInUk"
      popper-class="form-additional-info__field-popper"
    >
      <el-option
        label="Yes"
        value="true"
      />
      <el-option
        label="No"
      />
    </el-select>
  </div>
   
  <div
    class="form-additional-info__field"
  >
    <div
      class="form-additional-info__field-title required"
    >
      <label
        for="gitProfile"
      >
        Your Github Profile
      </label>
    </div>
     
    <div
      id="gitProfile"
      value=""
    />
  </div>
   
  <div
    class="form-additional-info__field"
  >
    <div
      class="form-additional-info__field-title required"
    >
      <label
        for="aboutYou"
      >
        About You
      </label>
    </div>
     
    <div
      id="aboutYou"
      placeholder="Let us know more about you. What are you in to?"
      resize="none"
      rows="9"
      type="textarea"
      value=""
    />
  </div>
   
  <div
    class="form-additional-info__actions"
  >
    <div
      text="NEXT"
      width="115"
    />
  </div>
</div>
`;

exports[`FormAdditionalInfo component gitProfile filled 1`] = `
<div
  class="form-additional-info"
>
  <form-header-stub
    class="form-additional-info__header"
    title="More About You"
  />
   
  <div
    class="form-additional-info__field"
  >
    <div
      class="form-additional-info__field-title required"
    >
      <label
        for="liveInUk"
      >
        Do you live in the UK?
      </label>
    </div>
     
    <el-select
      id="liveInUk"
      popper-class="form-additional-info__field-popper"
    >
      <el-option
        label="Yes"
        value="true"
      />
      <el-option
        label="No"
      />
    </el-select>
  </div>
   
  <div
    class="form-additional-info__field"
  >
    <div
      class="form-additional-info__field-title required"
    >
      <label
        for="gitProfile"
      >
        Your Github Profile
      </label>
    </div>
     
    <div
      id="gitProfile"
      value="test git profile"
    />
  </div>
   
  <div
    class="form-additional-info__field"
  >
    <div
      class="form-additional-info__field-title required"
    >
      <label
        for="aboutYou"
      >
        About You
      </label>
    </div>
     
    <div
      id="aboutYou"
      placeholder="Let us know more about you. What are you in to?"
      resize="none"
      rows="9"
      type="textarea"
      value=""
    />
  </div>
   
  <div
    class="form-additional-info__actions"
  >
    <div
      text="NEXT"
      width="115"
    />
  </div>
</div>
`;

exports[`FormAdditionalInfo component liveInUk 1`] = `
<div
  class="form-additional-info"
>
  <form-header-stub
    class="form-additional-info__header"
    title="More About You"
  />
   
  <div
    class="form-additional-info__field"
  >
    <div
      class="form-additional-info__field-title required"
    >
      <label
        for="liveInUk"
      >
        Do you live in the UK?
      </label>
    </div>
     
    <el-select
      id="liveInUk"
      popper-class="form-additional-info__field-popper"
    >
      <el-option
        label="Yes"
        value="true"
      />
      <el-option
        label="No"
      />
    </el-select>
  </div>
   
  <div
    class="form-additional-info__field"
  >
    <div
      class="form-additional-info__field-title required"
    >
      <label
        for="gitProfile"
      >
        Your Github Profile
      </label>
    </div>
     
    <div
      id="gitProfile"
      value=""
    />
  </div>
   
  <div
    class="form-additional-info__field"
  >
    <div
      class="form-additional-info__field-title required"
    >
      <label
        for="aboutYou"
      >
        About You
      </label>
    </div>
     
    <div
      id="aboutYou"
      placeholder="Let us know more about you. What are you in to?"
      resize="none"
      rows="9"
      type="textarea"
      value=""
    />
  </div>
   
  <div
    class="form-additional-info__actions"
  >
    <div
      text="NEXT"
      width="115"
    />
  </div>
</div>
`;
